### PR TITLE
Added superfluous optional notification

### PR DESCRIPTION
**Pull request by JoelBergstrand**
Notification from [OPTIONAL CALL PR](https://github.com/neo-technology/neo4j/pull/26755).

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-175.surge.sh

Updated pages:
- [Changes to status codes per Neo4j version](https://neo4j-docs-status-codes-175.surge.sh/status-codes/5/changelogs)
- [List of all server notification codes](https://neo4j-docs-status-codes-175.surge.sh/status-codes/5/notifications/all-notifications)

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 1]
```diff
@@ -3544,6 +3544,30 @@ RETURN *
 ======
 =====
 
+[#_neo_clientnotification_statement_superfluousoptional]
+=== Superfluous Optional
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3548]:
```suggestion
=== Superfluous optional
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 2]
```diff
@@ -3544,6 +3544,30 @@ RETURN *
...
+
+.Notification details
+[cols="<1s,<4"]
+|===
+|Neo4j code
+m|Neo.ClientNotification.Statement.SuperfluousOptional
+|Title
+a|The use of OPTIONAL is superfluous when `CALL` is a unit subquery or a void procedure.
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3556]:
```suggestion
a|The use of `OPTIONAL`  is superfluous when `CALL` is a unit subquery or a void procedure.
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 3]
```diff
@@ -3544,6 +3544,30 @@ RETURN *
...
+|Description
+|The use of OPTIONAL is superfluous when `CALL` is a unit subquery or a void procedure.
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3558]:
```suggestion
|The use of `OPTIONAL` is superfluous when `CALL` is a unit subquery or a void procedure.
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 4]
```diff
@@ -3544,6 +3544,30 @@ RETURN *
...
+|Category
+m|GENERIC
+|GQLSTATUS code
+m|03N61
+|Status description
+a|info: superfluous optional. The use of optional is superfluous when `CALL` is a unit subquery or a void procedure.
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3564]:
```suggestion
a|info: superfluous optional. The use of `OPTIONAL` is superfluous when `CALL` is a unit subquery or a void procedure.
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 5]
```diff
@@ -3544,6 +3544,30 @@ RETURN *
...
+|Classification
+m|GENERIC
+|SeverityLevel
+m|INFORMATION
+|===
+
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3570]: Could you also add an example of when this notification will be returned and a possible solution if such is available?

**Review by renetapopova — COMMENTED** [review]
Thanks, @JoelBergstrand. It looks good. I added a small suggestion and a comment.

File: modules/ROOT/pages/notifications/all-notifications.adoc (hunk 2 above)
Comment (renetapopova) [reply modules/ROOT/pages/notifications/all-notifications.adoc:3556]: This should also be changed in the code if we change it here.

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-175.surge.sh

Updated pages:
- [Changes to status codes per Neo4j version](https://neo4j-docs-status-codes-175.surge.sh/status-codes/5/changelogs)
- [List of all server notification codes](https://neo4j-docs-status-codes-175.surge.sh/status-codes/5/notifications/all-notifications)

**Comment by JoelBergstrand** [comment]
Thanks @renetapopova for the comments. We decided to split the notification in two, I've added examples two both of them and capitalized. Would appreciate you comments on the wording of the notifications and descriptions.

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 6]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+======
+Query::
++
+[source,cypher]
+----
+OPTIONAL CALL db.createLabel("A")
+----
+
+Description of the returned code::
+Optional is redundant in the case of void procedures. The use of `OPTIONAL` on void procedures have no effect and can be removed.
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3584]: Is this description the same as defined in the monorepo? I couldn't find it in the code.

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 7]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+
+Suggestions for improvement::
+If the intended behavior of the query is to use a void procedure the `OPTIONAL` keyword can be removed without impacting the query.
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3587]:
```suggestion
If the intended behavior of the query is to use a void procedure, the `OPTIONAL` keyword can be removed without impacting the query.
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 8]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
++
+[source,cypher]
+----
+CALL db.createLabel("A")
+----
+======
+=====
+
+[#_neo_clientnotification_statement_redundantoptionalsubquery]
+=== Redundant Optional Subquery
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3597]:
```suggestion
=== Redundant optional subquery
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 9]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+[source,cypher]
+----
+UNWIND [1, 2, 3] AS x
+OPTIONAL CALL (x) {
+   CREATE({i:x})
+}
+----
+
+Description of the returned code::
+Optional is redundant in the case of a unit subquery. The use of `OPTIONAL` on unit subqueries have no effect and can be removed.
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3636]: Same here. Is this description the same as defined in the monorepo? I couldn't find it in the code.

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 10]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+
+Suggestions for improvement::
+If the intended behavior of the query is for the subquery not to return any values the `OPTIONAL` keyword can be removed without impacting the query.
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3639]:
```suggestion
If the intended behavior of the query is for the subquery not to return any values, the `OPTIONAL` keyword can be removed without impacting the query.
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 11]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+m|03n62
+|status description
+a|info: redundant optional subquery. The use of `OPTIONAL` is redundant as `CALL` is a unit subquery.
+|classification
+m|generic
+|severitylevel
+m|information
+|===
+
+.Redundant use of optional on a call subquery.
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3620]:
```suggestion
.Redundant use of `OPTIONAL` in a `CALL` subquery
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 12]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+----
+CALL db.createLabel("A")
+----
+======
+=====
+
+[#_neo_clientnotification_statement_redundantoptionalsubquery]
+=== Redundant Optional Subquery
+
+.notification details
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3599]:
```suggestion
.Notification details
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 13]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+[cols="<1s,<4"]
+|===
+|neo4j code
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3602]:
```suggestion
|Neo4j code
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 14]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+m|neo.clientnotification.statement.redundantoptionalsubquery
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3603]:
```suggestion
m|Neo.ClientNotification.Statement.RedundantOptionalSubquery
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 15]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+|title
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3604]:
```suggestion
|Title
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 16]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+a|The use of `OPTIONAL` is redundant when `CALL` is a unit subquery.
+|description
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3606]:
```suggestion
|Description
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 17]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+|The use of `OPTIONAL` is redundant as `CALL` is a unit subquery.
+|category
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3608]:
```suggestion
|Category
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 18]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+m|generic
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3609]:
```suggestion
m|GENERIC
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 19]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+|gqlstatus code
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3610]:
```suggestion
|GQLSTATUS code
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 20]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+m|03n62
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3611]:
```suggestion
m|03N62
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 21]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+|status description
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3612]:
```suggestion
|Status description
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 22]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+a|info: redundant optional subquery. The use of `OPTIONAL` is redundant as `CALL` is a unit subquery.
+|classification
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3614]:
```suggestion
|Classification
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 23]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+m|generic
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3615]:
```suggestion
m|GENERIC
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 24]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+|severitylevel
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3616]:
```suggestion
|SeverityLevel
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 25]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+m|information
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3617]:
```suggestion
m|INFORMATION
```

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 26]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+|===
+
+.Redundant use of optional on a procedure call.
+[.tabbed-example]
+=====
+[.include-with-neo4j-code]
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3574]: You must also add a tab example for the GQLSTATUS code, similar to the rest of the notifications.

File: modules/ROOT/pages/notifications/all-notifications.adoc [hunk 27]
```diff
@@ -3544,6 +3544,110 @@ RETURN *
...
+|status description
+a|info: redundant optional subquery. The use of `OPTIONAL` is redundant as `CALL` is a unit subquery.
+|classification
+m|generic
+|severitylevel
+m|information
+|===
+
+.Redundant use of optional on a call subquery.
+[.tabbed-example]
```
Comment (renetapopova) [inline modules/ROOT/pages/notifications/all-notifications.adoc:3621]: This one also needs a tab example for the GQLSTATUS code, similar to the rest of the notifications.

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-175.surge.sh

Updated pages:
- [Changes to status codes per Neo4j version](https://neo4j-docs-status-codes-175.surge.sh/status-codes/5/changelogs)
- [List of all server notification codes](https://neo4j-docs-status-codes-175.surge.sh/status-codes/5/notifications/all-notifications)

File: modules/ROOT/pages/notifications/all-notifications.adoc (hunk 26 above)
Comment (JoelBergstrand) [reply modules/ROOT/pages/notifications/all-notifications.adoc:3574]: I think I've misunderstood the format. I'm pushing some changes to align with the rest of the notifications. I'll make sure that the notifications in the monorepo correspond to what is written here, in substance and formatting.

File: modules/ROOT/pages/notifications/all-notifications.adoc (hunk 27 above)
Comment (JoelBergstrand) [reply modules/ROOT/pages/notifications/all-notifications.adoc:3621]: I think I've misunderstood the format. I'm pushing some changes to align with the rest of the notifications. I'll make sure that the notifications in the monorepo correspond to what is written here, in substance and formatting.

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-status-codes-175.surge.sh

Updated pages:
- [Changes to status codes per Neo4j version](https://neo4j-docs-status-codes-175.surge.sh/status-codes/5/changelogs)
- [List of all server notification codes](https://neo4j-docs-status-codes-175.surge.sh/status-codes/5/notifications/all-notifications)

**Comment by JoelBergstrand** [comment]
Good morning @renetapopova! Do you have any additional comments on this addition to the docs?

**Pull request merged**